roxie/ccd/ccdserver.cpp

```diff
@@ -11273,10 +11270,25 @@ class CRoxieServerXmlWriteActivity : public CRoxieServerDiskWriteActivity
         desc->queryProperties().setProp("@format","utf8n");
         desc->queryProperties().setProp("@rowTag",rowTag.get());
         desc->queryProperties().setProp("@kind", (kind==TAKjsonwrite) ? "json" : "xml");
-        if (headerLength)
-            desc->queryProperties().setPropInt("@headerLength", headerLength);
-        if (footerLength)
-            desc->queryProperties().setPropInt("@footerLength", footerLength);
+        if (kind==TAKjsonwrite)
+        {
+            if (headerLength)
+                desc->queryProperties().setPropInt("@headerLength", headerLength);
+            if (footerLength)
+                desc->queryProperties().setPropInt("@footerLength", footerLength);
+        }
+        else
+        {
+            IPartDescriptorIterator *parts = desc->getIterator();
+            if (parts->first())
+            {
+                do {
+                    IPartDescriptor &part = parts->query();
+                    part.queryProperties().setPropInt("@headerLength", headerLength);
+                    part.queryProperties().setPropInt("@footerLength", footerLength);
```

Comment:
@jakesmith 
@AttilaVamos 
After looking closer I think when ECL writes the file both json and xml should save the header lengths at the FILE, not the PART level.  ECL will never create different headerlengths per part.

When we read in the lengths, we can check the file level first, only if its not there would we check per part.

Also I think I should look at fixing JSON to always write the header lengths, even if 0.